src/fstab-generator/fstab-generator.c

```diff
@@ -327,14 +328,28 @@ static int add_mount(
             mount_point_ignore(where))
                 return 0;
 
+        r = fstab_extract_values(opts, "x-systemd.wanted-by", &wanted_by);
+        if (r < 0)
+                return r;
+
+        r = fstab_extract_values(opts, "x-systemd.required-by", &required_by);
+        if (r < 0)
+                return r;
+
         if (path_equal(where, "/")) {
                 if (flags & NOAUTO)
                         log_warning("Ignoring \"noauto\" for root device");
                 if (flags & NOFAIL)
                         log_warning("Ignoring \"nofail\" for root device");
                 if (flags & AUTOMOUNT)
                         log_warning("Ignoring automount option for root device");
+                if (!strv_isempty(wanted_by))
+                        log_warning("Ignoring \"x-systemd.wanted-by\" for root device");
+                if (!strv_isempty(required_by))
+                        log_warning("Ignoring \"x-systemd.required-by\" for root device");
 
+                strv_clear(required_by);
+                strv_clear(wanted_by);
```

Comment:
hmm, just use strv_free():

```c
required_by = strv_free(required_by);
wanted_by = strv_free(wanted_by);
```

We probbaly should get rid of `strv_clear()` altogether. That function frees all elements of the string array, but not the array itself, which is a strange operation. Let's just use strv_free() here which also frees the array